### PR TITLE
[DatePipe] add deprecation warnings for DataPipes that will solely exist in TorchData

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65827

Differential Revision: [D31272794](https://our.internmc.facebook.com/intern/diff/D31272794)

cc @VitalyFedyunin @ejguan @NivekT

### CODE REVIEW

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

File: torch/utils/data/datapipes/utils/common.py [hunk 1]
```diff
@@ -63,3 +63,8 @@ def validate_pathname_binary_tuple(data):
         raise TypeError("pathname binary tuple should have string type pathname, but got {}".format(type(data[0])))
     if not isinstance(data[1], BufferedIOBase):
         raise TypeError("pathname binary tuple should have BufferedIOBase based binary type, but got {}".format(type(data[1])))
+
+# Warns user that the DataPipe has been moved to TorchData and will be removed from `torch`
+def deprecation_warning_torchdata(name):
+    warnings.warn(f"""{name} is deprecated and will be removed from the package `torch`. Please import it """
+                   """from the new package TorchData: https://github.com/pytorch/data""", DeprecationWarning)
```
Comment (ejguan) [inline torch/utils/data/datapipes/utils/common.py]: nit: Maybe also talk about functional API for these DataPipe also exists in TorchData.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

**Review by ejguan — APPROVED** [review]
LGTM

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31272794).

**Comment by ejguan** [comment]
Same as the another PR. Could we add this PR to release 1.10?

**Pull request closed without merging**